src/components/language/language.service.ts

```diff
@@ -200,6 +202,29 @@ export class LanguageService {
     ];
   };
 
+  protected async checkPermission(
+    label: string,
+    id: string,
+    session: ISession
+  ) {
+    if (!session.userId) {
+      session.userId = this.config.anonUser.id;
+    }
+
+    const result = await this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .call(matchUserPermissions, label, id)
+      .return('permList, node')
+      .first();
+
+    if (!result) {
+      throw new ForbiddenException(
+        'User does not have permission to read a language'
+      );
```

Comment:
Read or update?

Reply:
Yeah, update!

Reply:
You aren't checking for either of these permissions though. How do you know which one?

Reply:
Same ^

Reply:
So I have to check the permissions(`canEdit`) fields one by one? I just checked the `permList` and if it's empty array, it throws an exception.

Reply:
@zkhin try these changes on lines 490+ 498. That should do it. Let me know.
![image](https://user-images.githubusercontent.com/1165460/90673350-3b987e80-e21d-11ea-8476-94c448e2ac9d.png)

Reply:
After only these changes, we will get `Could not find object` 404 exceptions if the user does not have permission. Is this fine? And I don't need to check permissions before updating `Language`?

Reply:
yes that is fine for now. after mvp we will refactor to give the FE better errors.

Reply:
Okay, cool